### PR TITLE
CCO-233: Add Azure AD Workload Identity doc.

### PR DESCRIPTION
**Pull request by abutcher**

### CODE REVIEW

File: docs/azure_workload_identity.md [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+Without Azure AD Workload Identity, an Azure credentials secret includes an `azure_client_secret` which is essentially a password used to authenticate with Azure AD for the identity identified by `azure_client_id`. This `azure_client_secret` needs to be kept secure and is not rotated automatically.
+
+```yaml
+apiVersion: v1
+data:
+  azure_client_id: <client id>
+  azure_client_secret: <client secret>
+  azure_region: <region>
+  azure_subscription_id: <subscription id>
+  azure_tenant_id: <tenant id>
```
Comment (jianping-shu) [inline docs/azure_workload_identity.md:25]: Maybe we shall add azure_resource_prefix and azure_resourcegroup here, to keep the azure credential secret format consistent in CCO doc.

File: docs/azure_workload_identity.md [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+2. Extract Azure CredentialsRequests objects from the release image.
+
+   ```
+   oc adm release extract --cloud=azure --credentials-requests $RELEASE_IMAGE --to=./credreqs
+   ```
+
+3. Extract the `openshift-install` and `ccoctl` binaries from the release image.
+
+   ```
+   oc adm release extract --command=openshift-install $RELEASE_IMAGE
```
Comment (huangmingxia) [inline docs/azure_workload_identity.md:64]: Here we need to confirm that if the pull-secret is not specified, can not to extract openshift-install:
`[hmx@fedora auth]$ oc adm release extract --command=openshift-install registry.ci.openshift.org/ocp/release:4.14.0-0.nightly-2023-07-11-092038`
```
Warning: the default reading order of registry auth file will be changed from "${HOME}/.docker/config.json" to podman registry config locations in the future version of oc. "${HOME}/.docker/config.json" is deprecated, but can still be used for storing credentials as a fallback. See https://github.com/containers/image/blob/main/docs/containers-auth.json.5.md for the order of podman registry config locations.
Warning: the default reading order of registry auth file will be changed from "${HOME}/.docker/config.json" to podman registry config locations in the future version of oc. "${HOME}/.docker/config.json" is deprecated, but can still be used for storing credentials as a fallback. See https://github.com/containers/image/blob/main/docs/containers-auth.json.5.md for the order of podman registry config locations.
error: unable to read image quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:7c76f09717ef38a6692b100784ae52c3f6e87a241a3635754f9189957640d79a: unauthorized: access to the requested resource is not authorized
```

I usually add pull-secret for authentication.
```
[hmx@fedora auth]$ oc adm release extract --command=openshift-install registry.ci.openshift.org/ocp/release:4.14.0-0.nightly-2023-07-11-092038 -a ~/.pull-secret

[hmx@fedora auth]$ ll | grep openshift-install
-rwxr-xr-x. 1 hmx hmx 632181720  7月11日 13:32 openshift-install

```

Also, is it necessary to extract oc.

**Comment by huangmingxia** [comment]
@abutcher I recorded the steps I took to install one Azure Workload Identity cluster in the test case[ OCP-64892](https://polarion.engineering.redhat.com/polarion/#/project/OSE/workitems?query=OCP-64892). Perhaps we need to make some adjustments to the installation steps.

File: docs/azure_workload_identity.md [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+   This command should return a secret not found error.
+
+   ```yaml
+   oc get secrets -n kube-system azure-credentials
+   ```
+
+2. Verify that components are assuming the `azure_client_id` specified in the secret manifests, instead of credentials passed through by the Cloud Credential Operator. The secret displayed should not contain an `azure_client_secret` key and will instead contain an `azure_federated_token_file` key.
+   ```yaml
+   oc get secrets -n openshift-image-registry cloud-credentials -o yaml
+   ```
```
Comment (huangmingxia) [inline docs/azure_workload_identity.md:146]: namespace openshift-image-registry  did not have cloud-credentials secrets. 
```
[hmx@fedora auth]$  oc get secrets -n openshift-image-registry cloud-credentials -o yaml
Error from server (NotFound): secrets "cloud-credentials" not found
```
```
[hmx@fedora auth]$  oc get secrets -n openshift-image-registry
NAME                                              TYPE                                  DATA   AGE
builder-dockercfg-m9pbh                           kubernetes.io/dockercfg               1      5h12m
builder-token-f8pwt                               kubernetes.io/service-account-token   4      5h12m
cluster-image-registry-operator-dockercfg-bbzh5   kubernetes.io/dockercfg               1      5h12m
cluster-image-registry-operator-token-9z9sw       kubernetes.io/service-account-token   4      5h12m
default-dockercfg-t7bjv                           kubernetes.io/dockercfg               1      5h12m
default-token-s7wkc                               kubernetes.io/service-account-token   4      5h12m
deployer-dockercfg-h94b6                          kubernetes.io/dockercfg               1      5h12m
deployer-token-4hb78                              kubernetes.io/service-account-token   4      5h12m
image-registry-operator-tls                       kubernetes.io/tls                     2      5h15m
image-registry-private-configuration              Opaque                                0      5h7m
image-registry-tls                                kubernetes.io/tls                     2      5h7m
installation-pull-secrets                         kubernetes.io/dockerconfigjson        1      5h7m
installer-cloud-credentials                       Opaque                                5      5h23m
node-ca-dockercfg-9lwzr                           kubernetes.io/dockercfg               1      5h12m
node-ca-token-2p4bb                               kubernetes.io/service-account-token   4      5h12m
pruner-dockercfg-r8jn6                            kubernetes.io/dockercfg               1      5h12m
pruner-token-rd264                                kubernetes.io/service-account-token   4      5h12m
registry-dockercfg-s49pp                          kubernetes.io/dockercfg               1      5h7m
registry-token-mh4md                              kubernetes.io/service-account-token   4      5h7m
```

Comment (huangmingxia) [reply docs/azure_workload_identity.md:146]: It should be the installer-cloud-credentials secret.
```
$  oc get secrets -n openshift-image-registry installer-cloud-credentials -o yaml
apiVersion: v1
data:
  azure_client_id: xxx
  azure_federated_token_file: xxx
  azure_region: xxx
  azure_subscription_id: xxx
  azure_tenant_id: xxx
kind: Secret
metadata:
  creationTimestamp: "2023-07-14T04:07:38Z"
  name: installer-cloud-credentials
  namespace: openshift-image-registry
  resourceVersion: "1008"
  uid: d41ad220-350d-4950-ac83-9b12d4f3c82a
type: Opaque
```

File: docs/azure_workload_identity.md (hunk 2 above)
Comment (abutcher) [reply docs/azure_workload_identity.md]: [cf20a62](https://github.com/openshift/cloud-credential-operator/pull/566/commits/cf20a62af76fa90b18b38ed32aef4d9fd0f14d92) adds a step for obtaining `oc`.

**Comment by huangmingxia** [comment]
LGTM.

**Review by jstuever — APPROVED** [review]
/lgtm

**Pull request merged**